lib/src/function-registry.ts

```diff
@@ -119,3 +126,46 @@ export class FunctionRegistry<sync extends 'sync' | 'async'> {
     }
   }
 }
+
+/**
+ * Implements the simplification algorithm for custom function return values.
+ *  {@link https://github.com/sass/sass/blob/main/spec/types/calculation.md#simplifying-a-calculationvalue}
+ */
+function simplify(value: unknown): unknown {
+  if (value instanceof SassCalculation) {
+    const simplifiedArgs = value.arguments.map(
+      simplify
+    ) as List<CalculationValue>;
+    if (value.name === 'calc') {
+      return simplifiedArgs.get(0);
+    }
+    if (value.name === 'clamp') {
+      if (simplifiedArgs.size !== 3) {
+        throw new Error('clamp() requires exactly 3 arguments.');
+      }
+      return SassCalculation.clamp(
+        simplifiedArgs.get(0) as CalculationValue,
+        simplifiedArgs.get(1),
+        simplifiedArgs.get(2)
+      );
+    }
+    if (value.name === 'min') {
+      return SassCalculation.min(simplifiedArgs);
+    }
+    if (value.name === 'max') {
+      return SassCalculation.max(simplifiedArgs);
+    }
+    // @ts-expect-error: Constructor is private, but we need a new instance here
+    return new SassCalculation(value.name, simplifiedArgs);
```

Comment:
@nex3 I started implementing `simplify()`, but I realized that returning `SassCalculation` instances here will not automatically simplify the results back into primitive types as is the case in `dart-sass`. I presume we don't want to re-implement all the [logic](https://github.com/sass/dart-sass/blob/262961cb576d46f5518097c2c79c3e0dfb0e384d/lib/src/value/calculation.dart#L135-L159) from min, max, and clamp here, right? How should I tackle this?

Reply:
The simplification should be handled by the embedded compiler. That is, the embedded host should send an _unsimplified_ calculation object back to the compiler, and then the compiler should use the existing Dart simplification logic to simplify the values on that end.